### PR TITLE
Change the way SELinux is applied for portnumbers 

### PR DESCRIPTION
**Pull request by ralfbosz**
When declaring serveral http_ports with different IP's, but the same portnumber a resource conflict would arise on the selinux-part. Changed the selinux::port to an "ensure_resource", hence fixing issue #120.

### CODE REVIEW

**Comment by traylenator** [comment]
Hi @ralfbosz the subject of the pull request could be more precise. It sounds a lot more alarming than it currently is.

**Comment by traylenator** [comment]
Tests failed due to failed download - rerunning.

File: manifests/http_port.pp [hunk 1]
```diff
@@ -51,12 +51,12 @@
   }
 
   if $facts['selinux'] == true {
-    selinux::port{"selinux port squid_port_t ${_port}":
+    ensure_resource('selinux::port', "selinux port squid_port_t ${_port}", {
```
Comment (traylenator) [inline manifests/http_port.pp]: Hi, Can a test (rspec or acceptance)  be added to confirm that adding two `http_ports` does indeed  not cause a problem.

Comment (ralfbosz) [reply manifests/http_port.pp]: Added rspec, discovered that previous rspecs for SELinux I created weren't working, fixed that too...

**Comment by traylenator** [comment]
@ralfbosz happy with that title? More useful in the eventual changelog I think.

**Comment by ralfbosz** [comment]
Adjusted the commit title...

**Comment by traylenator** [comment]
From the review docs:

> Write the Pull Request Title in a way that users can easily identify if they are impacted or not

**Comment by ralfbosz** [comment]
Seems ensure_resource is a legacy thing (error), could have known after the presentation yesterday from Alexander Fisher.

**Comment by ekohl** [comment]
@alexjfisher is `ensure_resource` a legacy thing? I thought `create_resources` was, but ensure is to avoid duplicate definitions.

File: manifests/http_port.pp [hunk 2]
```diff
@@ -50,13 +50,13 @@
     order   => "30-${order}",
   }
 
-  if $facts['os']['selinux'] == true {
-    selinux::port{"selinux port squid_port_t ${_port}":
+  if $facts['selinux'] == true {
```
Comment (ekohl) [inline manifests/http_port.pp]: Please continue to use the modern facts

```suggestion
  if $facts['os']['selinux'] == true {
```

**Comment by alexjfisher** [comment]
> @alexjfisher is `ensure_resource` a legacy thing? I thought `create_resources` was, but ensure is to avoid duplicate definitions.

`ensure_resource` is pretty hacky, but unlike `create_resources` I don't think there's a new Puppet 4+ alternative.

**Comment by alexjfisher** [comment]
@ralfbosz What's the status of this?  There's been interest on Slack in getting this merged.

**Comment by ralfbosz** [comment]
Fixed it...

**Review by bastelfreak — APPROVED** [review]

File: manifests/http_port.pp [hunk 3]
```diff
@@ -31,6 +31,7 @@
 #   When set to `true` creates https_port entries. Defaults to `false`.
 # @param order
 #   Order can be used to configure where in `squid.conf`this configuration section should occur.
+
```
Comment (ekohl) [inline manifests/http_port.pp:34]: I think I've seen tools break on this empty line

**Review by ekohl — CHANGES_REQUESTED** [review]
Minor detail inline, but overall :+1:

File: manifests/http_port.pp (hunk 3 above)
Comment (ralfbosz) [reply manifests/http_port.pp]: Copy past error, removed it.

Happy to help !

**Review by ekohl — APPROVED** [review]
Travis is still running, but the code looks good.

**Pull request merged**